src/test/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPJwtKeyByOpenIdConnectAuthenticatorTest.java

```diff
@@ -108,20 +111,94 @@ public void testRoles() throws Exception {
 		Assert.assertEquals(TestJwts.TEST_ROLES, creds.getBackendRoles());
 	}
 
+
 	@Test
 	public void testExp() throws Exception {
 		Settings settings = Settings.builder().put("openid_connect_url", mockIdpServer.getDiscoverUri()).build();
 
 		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
-
+		
 		AuthCredentials creds = jwtAuth.extractCredentials(
 				new FakeRestRequest(ImmutableMap.of("Authorization", TestJwts.MC_COY_EXPIRED_SIGNED_OCT_1),
 						new HashMap<String, String>()),
 				null);
 
 		Assert.assertNull(creds);
+	}	
+
+	@Test
+	public void testExpInSkew() throws Exception {
+		Settings settings = Settings.builder()
+			.put("openid_connect_url", mockIdpServer.getDiscoverUri())
+			.put("clockSkewToleranceSeconds", "10")
+			.build();
+
+		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
+
+		long expiringDate = System.currentTimeMillis()/1000-5;
+
+		JwtToken jwt_token = TestJwts.create(TestJwts.MCCOY_SUBJECT, TestJwts.TEST_AUDIENCE, 
+		TestJwts.ROLES_CLAIM, TestJwts.TEST_ROLES_STRING, 
+		JwtConstants.CLAIM_EXPIRY, expiringDate);
+
+		String token=TestJwts.createSigned(jwt_token, TestJwk.OCT_1);
+		AuthCredentials creds = jwtAuth.extractCredentials(
+				new FakeRestRequest(ImmutableMap.of("Authorization", "bearer "+token),
+						new HashMap<String, String>()),
+				null);
+
+		Assert.assertNotNull(creds);
 	}
 
+	@Test
+	public void testNbf() throws Exception {
```

Comment:
Nit: Consider renaming this test to something like "testTooEarlyWithoutSkew". Just something that gives you an idea what it does from the name alone because in the logs, it will only show the test class and name.

Reply:
the name of this test matches the already existing test "testExp"

Reply:
The name doesn't make sense even if you know the acronym, lets update all the names of the tests to describe what is being tested.